packages/client/src/modals/SuccessModal.js

```diff
@@ -0,0 +1,138 @@
+import React, { useContext, useEffect, useState } from "react";
+import { Web3Context } from "contexts/Web3";
+import { useClipboard } from "../hooks";
+import { Copy, ArrowUp, OpenNewTab, Close } from "../components/Svg";
+import { getFlowscanUrlForTransaction } from "utils";
+import { ASSET_TYPES } from "../constants/enums";
+import { COIN_TYPE_TO_META, CONTRACT_NAME_TO_COIN_TYPE } from "../constants/maps";
+
+const SuccessModal = ({ data, type, txID, onClose, safeName, safeAddress, safeOwners }) => {
+    const { getNFTReference } = useContext(Web3Context);
+    const clipboard = useClipboard();
+
+    const [imageURI, setImageURI] = useState();
+
+    useEffect(() => {
+        if (type === ASSET_TYPES.NFT) {
+            const getImageURL = async () => {
+                const result = await getNFTReference(data.recipientAddr, data.nftID);
+                setImageURI(result);
+            };
+            getImageURL().catch(console.error);
+        }
+    }, [type])
+
+    function getNFTName(collectionId) {
+        return collectionId.split(".")[2];
+    }
+
+    function getTokenName(vaultId) {
+        const tokenName = vaultId.split(".")[2];
+        return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]].displayName;
+    }
+
+    function getTokenIcon(vaultId) {
+        const tokenName = vaultId.split(".")[2];
+        return COIN_TYPE_TO_META[CONTRACT_NAME_TO_COIN_TYPE[tokenName]].icon;
+    }
+
+    function getSafeOwnerName(address) {
+        const owner = safeOwners.filter(owner => owner.address === address);
+        return owner && owner[0].name;
```

Comment:
This is for rendering recipient name, right? Maybe name it `getRecipientName` as recipient might not be an owner. We also need to look for "SavedAddress/ContactList" to check if there is a name for given address. @mannynotfound  is work on Contact page. If it's difficult to coordinate in this PR we can create a ticket to pick up later.

Reply:
Yeah, currently we reuse safe owners as contacts, so after Manny's work is done this can be updated

Reply:
Is the purpose here to do nullish checking? suggest using optional chain
```suggestion
        return owner?.name;
```

Reply:
Updated, and used contacts instead of safe owners